### PR TITLE
Support MySQL database user add and delete operation

### LINKED ISSUE #1055: As an operator, I can use the Service operator to manage the users within MySQL database
This user story covers the following:
The Service Operator can add users with roles to MySQL databases. 

**Acceptance Criteria**
[Update this once the owner starts looking at the user story]

- [ ] The service operator can create a user within a MySQL database with role name, username and password. 

- [ ] The service operator can update the user's password

- [ ] The service operator can delete a user within the MySQL database. 

**Tasks**
Reference this issue when you create tasks for this user story

### PR DESCRIPTION
**Pull request by buhongw7583c**
Closes #1055 

**What this PR does / why we need it**:
With this PR, enduser could use Azure Service Operator to manage users for MySQL database: 
-- Add a new user with defined privileges to MySQL Database on Azure
-- Modify an existing user with new privileges
-- Delete a user from the MySQL database on Azure

**Special notes for your reviewer**:
For testing: 
Use the sample yaml file to create a mysql server, mysql database and mysql firewall rule to at least ensure your local client access to the Azure resource. 

Use the sample yaml file azure_v1alpha1_mysqluser.yaml to add a new user with privileges. 
Note:
1.  so far only privileges but not roles could be assigned to a MySQL user on Azure. 
2. If no key vault was provisioned with the mysql server, use the default kube secret which could be derived from kubectl get secret. Usually the secret name is the same as the sever name. 

After executed the yaml file, use any tool (e.g. MySQL workbench) to connect to MySQL and verify the new user has been added with the defined privileges. 

image

Delete the newly added user using 'kubect delete mysqluser <mysqlusername>'. Note that the mysqlusername should be the name of the metadata but not the MySQL username. 

The MySQL user has been deleted. 

image

**How does this PR make you feel**:
image

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains tests

### CODE REVIEW

**Comment by buhongw7583c** [comment]
@melonrush13  Mel, this PR is ready for review as well. The same opening discussion as in postgresqluser: whether we would like to stop the reconcile of a delete procedure, when the IP address has no access to the server. When we reach the agreement, we can make the changes(or without change) for both PR.

File: config/samples/azure_v1alpha1_mysql_everything.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+apiVersion: azure.microsoft.com/v1alpha1
```
Comment (melonrush13) [inline config/samples/azure_v1alpha1_mysql_everything.yaml:1]: Great addition adding this file!

**Comment by melonrush13** [comment]
> @melonrush13 Mel, this PR is ready for review as well. The same opening discussion as in postgresqluser: whether we would like to stop the reconcile of a delete procedure, when the IP address has no access to the server. When we reach the agreement, we can make the changes(or without change) for both PR.
> 
> Again, thanks so much for taking care of the two PR with excellent comments and made so many efforts to verify!

Love your GIF on this Hong!! 😆  Let's go ahead and add functionality to successfully delete the user when an improper IP is provisioned just like PostgreSqlUser and (soon to be ([issue](https://github.com/Azure/azure-service-operator/issues/1112) again for reference)) AzureSqlUser :)

**Comment by buhongw7583c** [comment]
> > @melonrush13 Mel, this PR is ready for review as well. The same opening discussion as in postgresqluser: whether we would like to stop the reconcile of a delete procedure, when the IP address has no access to the server. When we reach the agreement, we can make the changes(or without change) for both PR.
> > Again, thanks so much for taking care of the two PR with excellent comments and made so many efforts to verify!
> 
> Love your GIF on this Hong!! 😆 Let's go ahead and add functionality to successfully delete the user when an improper IP is provisioned just like PostgreSqlUser and (soon to be ([issue](https://github.com/Azure/azure-service-operator/issues/1112) again for reference)) AzureSqlUser :)

@melonrush13 Mel, thanks for all the great comments and as discussed the postgreSQL PR, let's currently delete the mysqluser as well when the IP address has no access to server. Later, let's handle the same way as what AzureSQLUser does. :)

**Pull request closed without merging**